systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -2128,6 +2137,344 @@ void testKafkaOffsetsReplicationFactorHigherThanReplicas() {
+                .withMessageCount(MESSAGE_COUNT)
+                .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
+                .withSecurityProtocol(SecurityProtocol.PLAINTEXT)
+                .build();
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, kafkaReplicas, 1)
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withNewKafkaListenerExternalNodePort()
+                                //.withTls(false)
+                            .endKafkaListenerExternalNodePort()
+                            .withNewPlain()
+                            .endPlain()
+                        .endListeners()
+                        .withConfig(kafkaConfig)
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+        String userName = "alice";
+        KafkaUserResource.tlsUser(CLUSTER_NAME, userName).done();
+        waitFor("Wait for secrets became available", Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_GET_SECRETS,
+            () -> kubeClient().getSecret(userName) != null,
+            () -> LOGGER.error("Couldn't find user secret {}", kubeClient().listSecrets()));
+
+        String userName2 = "john";
+        KafkaUserResource.tlsUser(CLUSTER_NAME, userName2).done();
+        waitFor("Wait for secrets became available", Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_GET_SECRETS,
+                () -> kubeClient().getSecret(userName2) != null,
+                () -> LOGGER.error("Couldn't find user secret {}", kubeClient().listSecrets()));
+
+        basicExternalKafkaClientTls.setKafkaUsername(userName);
+        basicExternalKafkaClient.setKafkaUsername(userName2);
+
+        assertThrows(ExecutionException.class, () -> {
+            Future<Integer> failproducer = basicExternalKafkaClient.sendMessagesPlain(5000);
+            Future<Integer> failconsumer = basicExternalKafkaClient.receiveMessagesPlain(5000);
+
+            assertThat(failproducer.get(2, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+            assertThat(failconsumer.get(2, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+        });
+
+        Future<Integer> producer = basicExternalKafkaClientTls.sendMessagesTls();
+        Future<Integer> consumer = basicExternalKafkaClientTls.receiveMessagesTls();
+
+        assertThat(producer.get(2, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+        assertThat(consumer.get(2, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+
+        LOGGER.info("Updating listeners of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaListeners updatedKl = new KafkaListenersBuilder()
+                    .withNewKafkaListenerExternalNodePort()
+                        .withNewKafkaListenerAuthenticationTlsAuth()
+                        .endKafkaListenerAuthenticationTlsAuth()
```

Comment:
Why do you adding tls authentification?

Reply:
I need to perform various _harakiri_ with configuration to see, whether the cluster stays functional after dynamic changes.